openmc/dagmc.py

```diff
@@ -0,0 +1,624 @@
+        present in the DAGMC file.
+
+        .. versionadded:: 0.13.2
+    n_surfaces : int
+        The number of surfaces in the model.
+
+        .. versionadded:: 0.15
+    material_overrides : dict
+        A dictionary of material overrides. The keys are material name
+        strings and the values are Iterables of openmc.Material objects. If a 
+        material name is found in the DAGMC file, the material will be replaced
+        with the openmc.Material object in the value.
+    """
+    
+    def __init__(self,
+                 filename: cv.PathLike,
+                 universe_id=None,
+                 name='',
+                 auto_geom_ids=False,
+                 auto_mat_ids=False,
+                 mat_overrides=None):
+        super().__init__(universe_id, name)
+        # Initialize class attributes
+        self.filename = filename
+        self.auto_geom_ids = auto_geom_ids
+        self.auto_mat_ids = auto_mat_ids
+        self.material_overrides = mat_overrides
+        self._dagmc_cells = []
+
+    def __repr__(self):
+        string = super().__repr__()
+        string += '{: <16}=\t{}\n'.format('\tGeom', 'DAGMC')
+        string += '{: <16}=\t{}\n'.format('\tFile', self.filename)
+        return string
+
+    @property
+    def bounding_box(self):
+        with h5py.File(self.filename) as dagmc_file:
+            coords = dagmc_file['tstt']['nodes']['coordinates'][()]
+            lower_left_corner = coords.min(axis=0)
+            upper_right_corner = coords.max(axis=0)
+            return openmc.BoundingBox(lower_left_corner, upper_right_corner)
+
+    @property
+    def filename(self):
+        return self._filename
+
+    @filename.setter
+    def filename(self, val: cv.PathLike):
+        cv.check_type('DAGMC filename', val, cv.PathLike)
+        self._filename = input_path(val)
+
+    @property
+    def material_overrides(self):
+        return self._material_overrides
+
+    @material_overrides.setter
+    def material_overrides(self, val):
+        if val is None:
+            self._material_overrides = val
+            return
+        else:
+            cv.check_type('material overrides', val, dict)
+            for key, value in val.items():
+                self.add_material_override(key, value)
+
+    def add_material_override(self, key, overrides=None):
+        """Add a material override to the universe.
+
+        .. versionadded:: 0.15
+
+        Parameters
+        ----------
+        key : str
+            Material name or ID of the Cell to override
+        value : Iterable of Materials
+            Materials to replace the key with
+
+        """
+        keys = []
+        if isinstance(key, str):
+            if key not in self.material_names:
+                raise ValueError(
+                    f"Material name '{key}' not found in DAGMC file")
+            elif int(key) in self.cells:
+                keys = [self.cells[int(key)]]
```

Comment:
I don't think we should require that the key meet both of these conditions. It should be one or there other right?